Bolts/src/bolts/CancellationTokenSource.java

```diff
@@ -58,4 +138,10 @@ public String toString() {
         Integer.toHexString(hashCode()),
         Boolean.toString(isCancellationRequested()));
   }
+
+  private void throwIfClosed() {
+    if (closed) {
+      throw new IllegalStateException("Object already closed");
+    }
+  }
```

Comment:
I can add a note - currently this method is always called from within a synchronized block.